arangosh/Restore/RestoreFeature.cpp

```diff
@@ -1784,4 +1824,80 @@ Result RestoreFeature::getFirstError() const {
   return {TRI_ERROR_NO_ERROR};
 }
 
+void RestoreFeature::ProgressTracker::updateStauts(const std::string& filename,
+                                                   const RestoreFeature::CollectionStatus& status) {
+  {
+    std::unique_lock guard(_collectionStatesMutex);
+    _collectionStates[filename] = status;
+
+    if (_writeQueued) {
+      return;
+    }
+
+    _writeQueued = true;
+  }
+
+  {
+    std::unique_lock guard(_writeFileMutex);
+    VPackBuilder builder;
+    {
+      std::unique_lock guardState(_collectionStatesMutex);
+      _writeQueued = false;
+      VPackObjectBuilder object(&builder);
+
+      for (auto const& [filename, state] : _collectionStates) {
+        builder.add(VPackValue(filename));
+        state.toVelocyPack(builder);
+      }
+    }
+
+
+    LOG_DEVEL << "Continuation file = " << builder.toJson();
+    auto file = directory.writableFile("continue.json", true, 0, false);
+    file->spit(builder.toJson());
```

Comment:
This is unsafe as it is not an atomic write. We should write to a temporary file instead and then `rename` the file, which is indeed a atomic operation. Otherwise we could end up with a partially written file... which defeats its own purpose.

Reply:
We can use VelocyPackHelper::velocyPackToFile, which can do this.

Reply:
It can also sync the file to disk, which we probably also want.

Reply:
Nice! :+1: